ibc/client/types/height_test.go

```diff
@@ -0,0 +1,430 @@
+				RevisionNumber: 2,
+				RevisionHeight: 2,
+			},
+			expected: false,
+		},
+		{
+			name: "GT: height == other",
+			op:   "GT",
+			height: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			other: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			expected: false,
+		},
+		{
+			name: "GT: height > other",
+			op:   "GT",
+			height: &Height{
+				RevisionNumber: 2,
+				RevisionHeight: 2,
+			},
+			other: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			expected: true,
+		},
+		{
+			name: "GT: height < other (same revision number)",
+			op:   "GT",
+			height: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			other: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 2,
+			},
+			expected: false,
+		},
+		{
+			name: "GT: height > other (same revision number)",
+			op:   "GT",
+			height: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 2,
+			},
+			other: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			expected: true,
+		},
+		{
+			name: "GT: height > other (same revision height)",
+			op:   "GT",
+			height: &Height{
+				RevisionNumber: 2,
+				RevisionHeight: 1,
+			},
+			other: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			expected: true,
+		},
+		{
+			name: "GT: height < other (same revision height)",
+			op:   "GT",
+			height: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			other: &Height{
+				RevisionNumber: 2,
+				RevisionHeight: 1,
+			},
+			expected: false,
+		},
+		{
+			name: "GTE: height < other",
+			op:   "GTE",
+			height: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			other: &Height{
+				RevisionNumber: 2,
+				RevisionHeight: 2,
+			},
+			expected: false,
+		},
+		{
+			name: "GTE: height == other",
+			op:   "GTE",
+			height: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			other: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			expected: true,
+		},
+		{
+			name: "GTE: height > other",
+			op:   "GTE",
+			height: &Height{
+				RevisionNumber: 2,
+				RevisionHeight: 2,
+			},
+			other: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			expected: true,
+		},
+		{
+			name: "EQ: height < other",
+			op:   "EQ",
+			height: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			other: &Height{
+				RevisionNumber: 2,
+				RevisionHeight: 2,
+			},
+			expected: false,
+		},
+		{
+			name: "EQ: height == other",
+			op:   "EQ",
+			height: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			other: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			expected: true,
+		},
+		{
+			name: "EQ: height > other",
+			op:   "EQ",
+			height: &Height{
+				RevisionNumber: 2,
+				RevisionHeight: 2,
+			},
+			other: &Height{
+				RevisionNumber: 1,
+				RevisionHeight: 1,
+			},
+			expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
I did not go through all these tests (just the first few), but given this functionality, assuming it works.